client/src/main/java/io/pravega/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -0,0 +1,367 @@
+
+    /**
+     * Deletes the schema associated to the given version. Users should be very careful while using this API in production, 
+     * esp if the schema has already been used to write the data. 
+     * An implementation of the delete call is expected to be idempotent. The behaviour of delete schema API invocation 
+     * with the schema registry service is idempotent. 
+     * The service performs a soft delete of the schema. So getSchemaVersion with the version info will still return the schema.
+     * However, the schema will not participate in any compatibility checks once deleted.
+     * It will not be included in listing schema versions for the group using APIs like {@link SchemaRegistryClient#getSchemaVersions}
+     * or {@link SchemaRegistryClient#getGroupHistory} or {@link SchemaRegistryClient#getSchemas} or 
+     * {@link SchemaRegistryClient#getLatestSchemaVersion}
+     * If add schema is called again using this deleted schema will result in a new version being assigned to it upon registration. 
+     * 
+     * @param groupId Id for the group.
+     * @param schemaType schemaType that identifies the type of object the schema represents. This should be same as the 
+     *                   value specified in {@link SchemaInfo#type}. 
+     * @param version Version number which uniquely identifies schema for the schemaType within a group. 
+     * @throws ResourceNotFoundException if group is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    void deleteSchemaVersion(String groupId, String schemaType, int version) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Gets schema corresponding to the version. 
+     * 
+     * @param groupId Id for the group. 
+     * @param versionInfo Version which uniquely identifies schema within a group. 
+     * @return Schema info corresponding to the version info.
+     * @throws ResourceNotFoundException if group or version is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    SchemaInfo getSchemaForVersion(String groupId, VersionInfo versionInfo) throws ResourceNotFoundException, UnauthorizedException;
+
+    /**
+     * Gets schema corresponding to the version. 
+     * 
+     * @param groupId Id for the group. 
+     * @param schemaType schemaType as specified in the {@link SchemaInfo#type} while registering the schema. 
+     * @param version Version which uniquely identifies schema of schemaType within a group. 
+     * @return Schema info corresponding to the version info.
+     * @throws ResourceNotFoundException if group or version is not found. 
+     * @throws UnauthorizedException if the user is unauthorized.
+     */
+    SchemaInfo getSchemaForVersion(String groupId, String schemaType, int version) throws ResourceNotFoundException, UnauthorizedException;
```

Comment:
This method takes "Group", "type", and "version". The above method takes "Group", "type", "version", and "id". They both return a "SchemaInfo". So the above method could be implemented by calling this one, and just throwing away the ID field. Therefore it is redundant to have both of these methods. I suggest removing the one above.

Reply:
i can remove this API. it is just a convenience api not used in the data path. (same goes for the corresponding delete api).